gammapy/background/energy_offset_array.py

```diff
@@ -28,7 +33,7 @@ def __init__(self, energy, offset, data=None):
         self.energy = Quantity(energy, 'TeV')
         self.offset = Angle(offset, 'deg')
         if data is None:
-            self.data = np.zeros((len(energy) - 1, len(offset) - 1))
+            self.data = Quantity(np.zeros((len(energy) - 1, len(offset) - 1)), "u")
```

Comment:
This is to be sure that whatever the method used to fill the data (fill_events or read) it will be a Quantity object. I need it for the evaluate method if I call self.myData.value